operator/operator_server.go

```diff
@@ -0,0 +1,119 @@
+package operator
+
+// since operator is a serverless component on its own, its associated server
+// is only used for hosting the /metrics, /livez and /readyz endpoints
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/grafana/grafana-app-sdk/health"
+)
+
+type OperatorServer struct {
+	Checks              []health.Check
+	Mux                 *http.ServeMux
+	Port                int
+	HealthCheckInterval time.Duration
+
+	// access to healthCheckErr is guarded with healthCheckErrMu
+	healthCheckErr   error
+	healthCheckErrMu sync.RWMutex
+}
+
+func (s *OperatorServer) RegisterMetricsHandler(handler http.Handler) {
+	s.Mux.Handle("/metrics", handler)
+}
+
+func (s *OperatorServer) registerHealthHandlers() error {
+	s.Mux.Handle("/livez", http.HandlerFunc(func(w http.ResponseWriter, _ *http.Request) {
+		_, _ = w.Write([]byte("ok"))
+		return
+	}))
+	s.Mux.Handle("/readyz", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if err := s.cachedChecks(); err != nil {
+			w.WriteHeader(http.StatusInternalServerError)
+			_, _ = w.Write([]byte("readiness check failed: " + err.Error()))
+			return
+		}
+		_, _ = w.Write([]byte("ok"))
+		return
+	}))
+	return nil
+}
+
+func (s *OperatorServer) Run(ctx context.Context) error {
```

Comment:
My async go is pretty novice. Please shred it with your focused eyes if need be.